chart/templates/etcd-statefulset.yaml

```diff
@@ -0,0 +1,180 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: etcd-{{ .Values.role }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: etcd-statefulset
+    role: {{ .Values.role }}
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  serviceName: etcd-{{.Values.role}}
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app: etcd-statefulset
+      role: {{ .Values.role }}
+  template:
+    metadata:
+      annotations:
+        checksum/configmap-etcd-bootstrap: {{ include (print $.Template.BasePath "/etcd-bootstrap-configmap.yaml") . | sha256sum }}
+{{- if .Values.podAnnotations }}
+{{ toYaml .Values.podAnnotations | indent 8 }}
+{{- end }}
+      labels:
+        app: etcd-statefulset
+        role: {{ .Values.role }}
+    spec:
+      containers:
+      - name: etcd
+        image: {{ index .Values.images "etcd" }}
+        imagePullPolicy: IfNotPresent
+        command:
+        - /var/etcd/bin/bootstrap.sh
+        readinessProbe:
+          httpGet:
+            path: /healthz
+            port: 8080
+          initialDelaySeconds: 15
+          periodSeconds: 5
+        livenessProbe:
+          exec:
+            command:
+            - /bin/sh
+            - -ec
+            - ETCDCTL_API=3
+            - etcdctl
+            - --config-file=/var/etcd/config/etcd.conf.yaml
+            - get
+            - foo
+          initialDelaySeconds: 15
+          periodSeconds: 5
+        ports:
+        - containerPort: 2380
+          name: serverport
+          protocol: TCP
+        - containerPort: 2379
+          name: clientport
+          protocol: TCP
+        resources:
+          requests:
+            cpu: 200m
+            memory: 500Mi
+          limits:
+            cpu: 750m
+            memory: 2560Mi
+        volumeMounts:
+        - name: etcd-{{ .Values.role }}
+          mountPath: /var/etcd/data
+        - name: etcd-bootstrap-sh
+          mountPath: /var/etcd/bin/
+        - name: etcd-config-file
+          mountPath: /var/etcd/config/
+{{- if .Values.tls }}
+        - name: ca-etcd
+          mountPath: /var/etcd/ssl/ca
+        - name: etcd-server-tls
+          mountPath: /var/etcd/ssl/server
+        - name: etcd-client-tls
+          mountPath: /var/etcd/ssl/client
+{{- end }}
+      - name: backup-restore
+        command:
+        - etcdbrctl
+        - server
+        - --schedule={{ .Values.backup.schedule }}
+{{- if eq .Values.backup.garbageCollectionPolicy "LimitBased" }}
+        - --max-backups={{ .Values.backup.maxBackups }}
+        - --garbage-collection-policy={{ .Values.gabageCollectionPolicy }}
+{{- end }}
+        - --data-dir=/var/etcd/data
```

Comment:
Please change this to `/var/etcd/data/new.etcd`.

Reply:
i assume i should also change https://github.com/gardener/etcd-backup-restore/pull/59/files/70c0f4117a02522de929169f47c2d7d62738f4e4#diff-39ba5fe174d707da3d38a1b4160f2284R69

Reply:
No. Mount path should be `/var/etcd/data`. Its correct. Reason here is, as per recent change restore operation happens at `<etcd-data-dir-path>.part` i.e in this case `/var/etcd/data/new.etcd.part`. Once restoration there is succeded, we rename it to actual `<etcd-data-dir>` i.e. `/var/etcd/data/new.etcd`.  So, if we change mount point to `/var/etcd/data/new.etcd` the rename operation will fail because of permission.

Reply:
Thanks for the confirmation.